### ISSUE TITLE
Paths with extraneous dot segments are not replaced correctly

### ISSUE THREAD

**Issue by keesse** [johnfactotum/foliate#953]
**Describe the bug**
Foliate 2.6.4 on Ubuntu 22.04 does not show images in [Darktable 4.0 manual](https://docs.darktable.org/usermanual/4.0/en/darktable_user_manual.epub), where FBReader does

**To Reproduce**
  In above epub, go to "Screen layout"
5. See error: no image, this is with all the images in this epub

**Expected behavior**
See the image

**Screenshots**

**Version:**
 - Foliate version: 2.6.4
 - OS/Distribution and version: Ubuntu 22.04
 - Installation method: apt (.deb file)

**Additional context**
![Foliate](https://user-images.githubusercontent.com/18166189/192139203-d09422fe-eb7b-487d-b151-8964508fa744.png)

**Comment by johnfactotum** [comment]
This is an Epub.js bug.

The problem is this. Take the page `/OEBPS/overview/user-interface/screen-layout.html` as an example.
1. The has an image with `src` set to `../../overview/user-interface/screen-layout/screen-layout.png`.
2. Epub.js sees the string `screen-layout/screen-layout.png`, and recognizes that this is the relative path to the image.
3. Epub.js replaces that string with a blob URL to the image.
4. The image's `src` is now `../../overview/user-interface/blob:[...]`, which is invalid.

So basically, it doesn't handle the case where a substring of the path is also a path to an resource.

More generally speaking, this is really an inherent problem with the method of replacing URLs used by Epub.js. It doesn't even parse the page when replacing. So it doesn't know that the `src` should be treated as one string.

I guess one way to fix this would be to require the match to start with `"`, `'`, `(`, etc. when replacing URLs.

**Comment by johnfactotum** [comment]
Thinking about it, I guess the problem is more that it doesn't replace *all* possible relative paths. It only replaces the shortest relative path. So it should generate all possible relative paths and replace them from longest to shortest.

**Comment by keesse** [comment]
@johnfactotum , thank you for the explanation

**Comment by johnfactotum** [comment]
Fixed in the new renderer (https://github.com/johnfactotum/foliate/issues/962), which parses the (X)HTML when replacing resources. So it recognizes the extra dot segments as parts of the URL and resolves them correctly.

**Comment by keesse** [comment]
@johnfactotum Thank you very much!
